invenio_circulation/transitions/base.py

```diff
@@ -109,7 +109,7 @@ def ensure_item_is_available(self, loan):
             msg = "Item not set for loan with pid '{}'".format(loan.id)
             raise TransitionConstraintsViolationError(description=msg)
 
-        if not is_item_available(loan['item_pid']):
+        if not item_can_circulate(loan['item_pid']):
```

Comment:
I think here is what @ntarocco was saying about keeping what is inside circulation module as `is_item_available_for_checkout` and `item_can_circulate` can be used for the config and the util functions. I would say the renaming is nice for the users of the module but internally the name can be kept as `is_item_available_for_checkout`. WDYT?